src/components/Navbar/index.spec.tsx

```diff
@@ -1,69 +1,104 @@
-    return { name };
+  const render = ({
+    _logOutFromServer = jest.fn(),
+    store = configureStore(),
+  }: RenderParams = {}) => {
+    // TODO: Use shallowUntilTarget()
+    // https://github.com/mozilla/addons-code-manager/issues/15
+    const root = shallow(<Navbar _logOutFromServer={_logOutFromServer} />, {
+      context: { store },
+    }).shallow();
+
+    return root;
+  };
+
+  const storeWithUser = (user = fakeUser) => {
+    const store = configureStore();
+
+    store.dispatch(userActions.loadCurrentUser({ user }));
+    return store;
   };
 
   describe('when a profile is provided', () => {
     it('renders a username', () => {
       const name = 'Bob';
-      const root = render({ profile: createProfile(name) });
+      const store = storeWithUser({ ...fakeUser, name });
+      const root = render({ store });
 
       expect(root.find(`.${styles.username}`)).toHaveText(name);
     });
 
     it('renders a log out button', () => {
-      const onLogOut = () => null;
-      const root = render({ onLogOut, profile: createProfile() });
+      const root = render({ store: storeWithUser() });
 
       expect(root.find(`.${styles.logOut}`)).toHaveLength(1);
     });
 
     it('does not render a log in button', () => {
-      const root = render({ profile: createProfile() });
+      const root = render({ store: storeWithUser() });
 
       expect(root.find(LoginButton)).toHaveLength(0);
     });
   });
 
   describe('when profile is null', () => {
     it('does not render a username', () => {
-      const root = render({ profile: null });
+      const root = render();
 
       expect(root.find(`.${styles.username}`)).toHaveLength(0);
     });
 
     it('renders a log in button', () => {
-      const root = render({ profile: null });
+      const root = render();
 
       expect(root.find(LoginButton)).toHaveLength(1);
     });
 
     it('does not render a log out button', () => {
-      const root = render({ profile: null });
+      const root = render();
 
       expect(root.find(`.${styles.logOut}`)).toHaveLength(0);
     });
   });
 
   describe('Log out button', () => {
-    it('calls onLogOut when clicked', () => {
-      const onLogOutMock = jest.fn();
-      const root = render({ onLogOut: onLogOutMock, profile: createProfile() });
+    it('calls logOutFromServer when clicked', () => {
+      const logOutFromServerMock = jest.fn();
+      const root = render({
+        _logOutFromServer: logOutFromServerMock,
+        store: storeWithUser(),
+      });
 
       root.find(`.${styles.logOut}`).simulate('click');
-      expect(onLogOutMock).toHaveBeenCalled();
+      expect(logOutFromServerMock).toHaveBeenCalled();
+    });
+
+    it('dispatches userActions.logOut when clicked', async () => {
+      const store = storeWithUser();
+      const dispatch = jest.spyOn(store, 'dispatch');
+
+      const root = render({ store });
+
+      await root.find(`.${styles.logOut}`).simulate('click');
```

Comment:
Enzyme does not provide any support for async event handlers so this probably isn't doing what you think it is